presto-main/src/main/java/com/facebook/presto/server/security/AuthenticationFilter.java

```diff
@@ -126,6 +142,50 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         }
     }
 
+    public HttpServletRequest mergeExtraHeaders(HttpServletRequest request, Principal principal)
+    {
+        List<ClientRequestFilter> clientRequestFilters = clientRequestFilterManager.getClientRequestFilters();
+
+        if (clientRequestFilters.isEmpty()) {
+            return request;
+        }
+
+        ImmutableMap.Builder<String, String> extraHeadersMapBuilder = ImmutableMap.builder();
+        Set<String> addedHeaders = new HashSet<>();
+
+        for (ClientRequestFilter requestFilter : clientRequestFilters) {
+            boolean headersPresent = requestFilter.getHeaderNames().stream()
+                    .allMatch(headerName -> request.getHeader(headerName) != null);
+
+            if (!headersPresent) {
+                ClientRequestFilterContext context = new ClientRequestFilterContext(principal);
+                Optional<Map<String, String>> extraHeaderValueMap = requestFilter.getExtraHeaders(context);
```

Comment:
Sorry, I realize now that the principal is per request.  Context objects typically are used to create the plugin.  So please revert this, and just add the principal as a plain old parameter to `getExtraHeaders`, no need for fancy parameterization.